tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -4258,9 +4332,16 @@ class HLSLExternalSource : public ExternalSemaSource {
   {
     DXASSERT_NOMSG(ULE != nullptr);
 
+    const bool isVkNamespace =
+        ULE->getQualifier() &&
+        ULE->getQualifier()->getKind() == NestedNameSpecifier::Namespace &&
+        ULE->getQualifier()->getAsNamespace()->getName() == "vk";
+
     // Intrinsics live in the global namespace, so references to their names
     // should be either unqualified or '::'-prefixed.
-    if (ULE->getQualifier() && ULE->getQualifier()->getKind() != NestedNameSpecifier::Global) {
+    // Exception: Vulkan-specific intrinsics live in the 'vk::' namespace.
+    if (!isVkNamespace && ULE->getQualifier() &&
+        ULE->getQualifier()->getKind() != NestedNameSpecifier::Global) {
```

Comment:
Minor: if you feel like it, you could combine some of the ULE->* checks to set another boolean for the global check.

Reply:
Done